engine/app/components/citizens_advice_components/input.rb

```diff
@@ -70,20 +71,26 @@ def hint?
       @hint.present?
     end
 
+    def general_id
+      return id if @id.present?
+
+      name
+    end
+
     def label_id
-      "#{name}-label"
+      "#{general_id}-label"
     end
 
     def input_id
-      "#{name}-input"
+      "#{general_id}-input"
```

Comment:
@leeky wanted to check what would make more sense in the case of form builder. I know that the rails way wouldn't have the `-input` at the end of the id, should we remove it? 🤔

Reply:
Rails doesn't really care about the HTML IDs, it's the `name` that it cares about.